armi/reactor/blueprints/componentBlueprint.py

```diff
@@ -310,3 +368,179 @@ def _insertDepletableNuclideKeys(c, blueprint):
+    """Specification for a single particle fuel type"""
+
+    item_type = _ParticleFuelLayer
+    key_attr = _ParticleFuelLayer.name
+    name = yamlize.Attribute(type=str)
+
+    def construct(self, blueprint, matMods):
+        """Produce a particle fuel instance that can be attached to a component"""
+        bounds = set()
+
+        layers = sorted(self.values(), key=lambda spec: spec.od)
+        spec = ParticleFuel(self.name)
+
+        prevInner = -inf
+        for layer in layers:
+            comp = layer.construct(blueprint, matMods)
+            innerDim = comp.getDimension("id")
+            if innerDim < 0:
+                raise ValueError(
+                    f"{layer.name} inner diameter must be non-negative, got "
+                    f"{innerDim}"
+                )
+            if innerDim <= prevInner:
+                inners = [ring.innerDiam for ring in layers]
+                raise ValueError(
+                    f"{self.name} has inconsistent inner diameters: not "
+                    f"increasing {inners}"
+                )
+            od = comp.getDimension("od")
+            if innerDim >= od:
+                raise ValueError(
+                    f"{layer.name} outer diameter must be greater than inner diameter. "
+                    f"Got {od} {innerDim}"
+                )
+
+            bounds.update((innerDim, od))
+            spec.add(comp)
+            prevInner = innerDim
+
+        nLayers = len(self)
+        if len(bounds) != nLayers + 1:
+            names = [m.name for m in spec]
+            raise ValueError(
+                f"{self.name} does not have consistent boundaries. Bounds: "
+                f"{sorted(bounds)}, compositions: {names}"
+            )
+
+        if not isclose(min(bounds), 0):
+            raise ValueError(
+                f"Particle fuel {self.name} does not start at radius of zero"
+            )
+
+        return spec
+
+
+class ParticleFuelKeyedList(yamlize.KeyedList):
+    """Keyed list to enable the ``particleFuel`` section
+
+    Structure
+    ---------
+
+    ..code:: yaml
+
+        particle fuel:
+            <specifier>:
+                <layer>:  # not strictly in increasing order
+                    material: <string>
+                    id: <float>
+                    od: <float>
+                    Tinput: <float>
+                    Thot: <float>
+                    flags: <optional list of strings>
+
+    Example
+    -------
+
+    ..code:: yaml
+
+        particle fuel:
+            TRISO:
+                kernel:
+                    material: UO2
+                    id: 0.6
+                    od: 0.61
+                    Tinput: 900
+                    Thot: 900
+                    flags: DEPLETABLE
+                buffer:
+                    material: SiC
+                    id: 0.6
+                    od: 0.62
+                    Tinput: 900
+                    Thot: 900
+                ...
```

Comment:
Just a high-level question here. What if we allowed "free" component definitions (e.g. not in blocks), along the lines of:

```yaml
components:
      kernel:
          material: UO2
          id: 0.6
          od: 0.61
          Tinput: 900
          Thot: 900
          flags: DEPLETABLE
      buffer:
          material: SiC
          id: 0.6
          od: 0.62
          Tinput: 900
          Thot: 900
      matrix:
          material: Graphite
          Tinput: 900
          Thot: 900
```
(ignoring the shape/dimension issue for the moment)

and then upgraded the blocks input to allow mixtures, along the lines of:

```yaml
        <block>:
            fuel:
                name:  fuelmatrix
                shape: Mixture
                mix: 
                  - comp: matrix
                    frac: 0.6
                  - comp: triso
                    frac: 0.4

```

And then have the input processing code take that and build a block that has mixes of those. Would that potentially be workable here? I'm thinking yes as long as you could mix matrix with TRISO, and let TRISO itself also be a composite (e.g. your ParticleFuel object below)

Reply:
Having free components (#505) is an awesome move. 

One potential issue with the mixture snippet above would be the `triso` component would need to be something more like a block as it would contain all the child layers. Or including multiple elements in the `mix` section
```yaml
mix:
  - comp: matrix
     frac: 0.6
  - comp: kernel
  - comp: buffer
```
Providing the fraction of kernel and buffer is not difficult, just using 40% times the volume percent of each layer in a sphere. But that's some additional steps required if whipping up an input by hand